openpype/modules/extra_addons/process_monitor/addon.py

```diff
@@ -0,0 +1,704 @@
+        if self._timers_manager_module is not None:
+            self._timers_manager_module.timer_stopped(self.id)
+
+        #TODO: needed? (should remove from other locations in code)
+        self._current_pid = None
+
+    def _add_running_process(self, data):
+        #self.log.warning("_add_running_process")
+
+        # TODO: should also check "username" (e.g. if background task started by server)
+        process_parameters = [
+            "name", #TODO: remove?
+            "environ",
+            "cmdline"
+        ]
+
+        #self.log.info("_add_running_process - current pid: {}".format(self._current_pid))
+        #TODO: keep current if none found?
+        #self._current_pid = None
+
+        new_pids = []
+        for process in psutil.process_iter(process_parameters):
+
+            pid = process.pid
+            if pid == 0:
+                continue
+
+            #self.log.info("cmdline : {}".format(process.info["cmdline"]))
+            if not process.info["cmdline"]:
+                continue
+            cmdline = [cmd.replace("\\", "/")
+                       for cmd in process.info["cmdline"]]
+
+            name = process.info["name"]
+            for application_name, project_application \
+                    in self._project_applications:
+                if project_application == cmdline[0]:
+                    process_cmd = cmdline[0]
+                    self.log.info("Found process: {}".format(name))
+                    break
+
+                elif project_application in cmdline[1:]:
+                    process_cmd = cmdline[1:].index(project_application)
+                    #TODO: REMOVE! (temp windows hack)
+                    name = project_application.split("\\")[-1].split("/")[-1]
+                    #name = ".".join(name.split(".")[:-1])
+                    self.log.info("Found subprocess: {}".format(name))
+                    break
+
+            else:
+                continue
+            #self.log.debug("PID: {}".format(pid))
+
+            ####
+            #TODO: get rid of 'name' stuff before?
+            name = application_name
+            ####
+
+            #TODO: could be done at start
+            # Check if parent process is running
+            ppid = process.ppid()
+            if ppid in self._running_processes:
+                self.log.warning("ppid '{}' running => ignore".format(ppid))
+                continue
+
+            #TODO: ok with "openpype"?
+            # => Not OK if started in CLI or via "run_tray" script
+            parent = process.parent()
+            if not parent or "openpype" not in parent.name():
+                continue
+
+            '''
+            self.log.info("PID: {}".format(pid))
+            self.log.info((
+                "_running_processes: {}"
+            ).format(self._running_processes))
+            '''
+
+            # Set current PID if matching context (can have more than 1 match)
+            environ = process.info["environ"]
```

Comment:
Can `psutils` catch changes of environments??

Reply:
I am not sure what you mean, but the environment caught by ```process.info``` seems to be updated for each new started application.